src/EfuncMsg.chpl

```diff
@@ -87,6 +87,46 @@ module EfuncMsg
                         var a = st.addEntry(rname, e.size, real);
                         a.a = cos(e.a);
                     }
+                    when "tan" {
+                        var a = st.addEntry(rname, e.size, real);
+                        a.a = tan(e.a);
```

Comment:
we should do some perf comparisons, but if this performs better, we should update all the efunc methods to do this. so something like
```chapel
var e = toSymEntry(gEnt,int);
ref ea = e.a;
select efunc {
    when "abs" {
        st.addEntry(rname, new shared SymEntry(abs(ea)));
    }
    when "log" {
        st.addEntry(rname, new shared SymEntry(log(ea)));
    }
    when "exp" {
        st.addEntry(rname, new shared SymEntry(exp(ea)));
    }
    when "sin" {
        st.addEntry(rname, new shared SymEntry(sin(ea)));
    }
    when "cos" {
        st.addEntry(rname, new shared SymEntry(cos(ea)));
    }
```
because this also cuts down on lines of code. that would be outside the scope of this PR though

Reply:
I would agree with this. Generally, it is more performant to create the SymEntry from an array rather than creating an empty SymEntry and populating the array. I agree we should do performance testing, but I definitely think @pierce314159 is correct here.

Reply:
This makes a lot of sense. I changed all of the `efunc` fxns to this. we can create an issue and look at performance later on.